### ISSUE TITLE
‘map_write_invalidate_region’ is not a member of ‘boost::compute::command_queue’

### ISSUE THREAD

**Issue by kaikai-sk** [Microsoft/LightGBM#1670]
when I install lightgbm-gpu use `pip install lightgbm --install-option=--gpu`
```
/tmp/pip-install-8apdfmkz/lightgbm/compile/src/treelearner/gpu_tree_learner.cpp: In member function ‘void LightGBM::GPUTreeLearner::AllocateGPUMemory()’:
/tmp/pip-install-8apdfmkz/lightgbm/compile/src/treelearner/gpu_tree_learner.cpp:275:103: error: ‘map_write_invalidate_region’ is not a member of ‘boost::compute::command_queue’
   ptr_pinned_gradients_ = queue_.enqueue_map_buffer(pinned_gradients_, boost::compute::command_queue::map_write_invalidate_region,
                                                                                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/pip-install-8apdfmkz/lightgbm/compile/src/treelearner/gpu_tree_learner.cpp:281:101: error: ‘map_write_invalidate_region’ is not a member of ‘boost::compute::command_queue’
   ptr_pinned_hessians_ = queue_.enqueue_map_buffer(pinned_hessians_, boost::compute::command_queue::map_write_invalidate_region,
                                                                                                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/pip-install-8apdfmkz/lightgbm/compile/src/treelearner/gpu_tree_learner.cpp:299:111: error: ‘map_write_invalidate_region’ is not a member of ‘boost::compute::command_queue’
   ptr_pinned_feature_masks_ = queue_.enqueue_map_buffer(pinned_feature_masks_, boost::compute::command_queue::map_write_invalidate_region,
                                                                                                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/pip-install-8apdfmkz/lightgbm/compile/src/treelearner/gpu_tree_learner.cpp:368:104: error: ‘map_write_invalidate_region’ is not a member of ‘boost::compute::command_queue’
     host4_ptrs[i] = (Feature4*)queue_.enqueue_map_buffer(host4_bufs[i], boost::compute::command_queue::map_write_invalidate_region,
                                                                                                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~
make[3]: *** [CMakeFiles/_lightgbm.dir/build.make:414: CMakeFiles/_lightgbm.dir/src/treelearner/gpu_tree_learner.cpp.o] Error 1
make[2]: *** [CMakeFiles/Makefile2:73: CMakeFiles/_lightgbm.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:85: CMakeFiles/_lightgbm.dir/rule] Error 2
make: *** [Makefile:164: _lightgbm] Error 2
```

**Comment by StrikerRUS** [comment]
@huanzhang12 please help

**Comment by kaikai-sk** [comment]
@huanzhang12  My os is fedora 28 server version. CUDA version is 9.2 .GPU is Tesla P40

**Comment by kaikai-sk** [comment]
@huanzhang12  I have resolved the problem by reinstall os,ubuntu 18.04.CUDA 9.0

**Comment by huanzhang12** [comment]
@kaikai-sk Glad to know it works for you!
